services/tasks-svc/internal/task/commands/v1/update_subtask.go

```diff
@@ -1,42 +1,39 @@
 package v1
 
 import (
-	"common"
 	"context"
+	"fmt"
 	"github.com/google/uuid"
-	"hwauthz"
 	"hwes"
-	"tasks-svc/internal/perm"
 	"tasks-svc/internal/task/aggregate"
 )
 
-type UpdateSubtaskCommandHandler func(ctx context.Context, taskID, subtaskID uuid.UUID, name *string) error
+type UpdateSubtaskCommandHandler func(ctx context.Context, taskID, subtaskID uuid.UUID, name *string, done *bool) error
 
-func NewUpdateSubtaskCommandHandler(as hwes.AggregateStore, authz hwauthz.AuthZ) UpdateSubtaskCommandHandler {
-	return func(ctx context.Context, taskID, subtaskID uuid.UUID, name *string) error {
-		userID, err := common.GetUserID(ctx)
-		if err != nil {
-			return err
-		}
+func NewUpdateSubtaskCommandHandler(as hwes.AggregateStore) UpdateSubtaskCommandHandler {
+	return func(ctx context.Context, taskID, subtaskID uuid.UUID, name *string, done *bool) error {
 		a, err := aggregate.LoadTaskAggregate(ctx, as, taskID)
 		if err != nil {
 			return err
 		}
 
-		user := perm.User(userID)
-		task := perm.Task(taskID)
-
-		check := hwauthz.NewPermissionCheck(user, perm.CanUserUpdateSubtaskOnTask, task)
-		if err = authz.Must(ctx, check); err != nil {
-			return err
+		currentSubtask, found := a.Task.Subtasks[subtaskID]
+		if !found {
+			return fmt.Errorf("Subtask with ID: %s not found on Task with ID: %s", subtaskID, taskID)
 		}
 
-		if name != nil {
+		if name != nil && *name != currentSubtask.Name {
 			if err := a.UpdateSubtaskName(ctx, subtaskID, *name); err != nil {
 				return err
 			}
 		}
 
+		if done != nil && *done != currentSubtask.Done {
```

Comment:
what if I want to undone a subtask?

Reply:
Assuming that `currentSubtask.Done = true` this can be done by calling the UpdateSubtask command with the param `done` set to `false`